src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -1295,8 +1378,10 @@ private Term prepareLimit(VariableSpecifications boundNames, Term.Raw limit,
             return prepLimit;
         }
 
-        private static void verifyOrderingIsAllowed(StatementRestrictions restrictions) throws InvalidRequestException
+        private static void verifyOrderingIsAllowed(StatementRestrictions restrictions, Map<ColumnMetadata, Ordering> orderingColumns) throws InvalidRequestException
```

Comment:
When users try to use not-ANN `ORDER BY` they get an error that says `ORDER BY with 2ndary indexes is not supported`. However, now we will support `ORDER BY ... ANN OF` with 2ndary indexes. 

So the message should probably say something like `ORDER BY with 2ndary indexes is not supported, except for ANN queries`, or something like that.